src/output.cpp

```diff
@@ -305,6 +318,39 @@ void TextOutput::attached_probes(uint64_t num_probes) const
     out_ << "Attaching " << num_probes << " probes..." << std::endl;
 }
 
+std::string TextOutput::tuple_to_str(BPFtrace &bpftrace,
+                                     const SizedType &ty,
+                                     const std::vector<uint8_t> &value) const
+{
+  bool first = true;
+  size_t offset = 0;
+  std::string ret;
+
+  ret += '(';
+
+  for (const SizedType &elemtype : ty.tuple_elems)
+  {
+    if (first)
+      first = false;
+    else
+      ret += ", ";
+
+    std::vector<uint8_t> elem_value(value.begin() + offset,
+                                    value.begin() + offset + ty.size);
```

Comment:
```suggestion
                                    value.begin() + offset + elemtype.size);
```
AddressSanitizer detects heap-buffer-overflow.

Reply:
Nice catch